packages/flutter_tools/lib/src/build_runner/build_runner.dart

```diff
@@ -204,10 +205,12 @@ class _BuildRunnerCodegenDaemon implements CodegenDaemon {
 // Sorts the builders by name and produces a hashcode of the resulting iterable.
 List<int> _produceScriptId(YamlMap builders) {
   if (builders == null || builders.isEmpty) {
-    return md5.convert(<int>[]).bytes;
+    return md5.convert(platform.version.codeUnits).bytes;
   }
   final List<String> orderedBuilders = builders.keys
     .cast<String>()
     .toList()..sort();
-  return md5.convert(orderedBuilders.join('').codeUnits).bytes;
+  return md5.convert(orderedBuilders
+    .followedBy(<String>[platform.version])
```

Comment:
nit: I don't know why, but for some reason I want the platform version to be at the beginning of the list. Maybe it's because it's more stable than the list of builders.

Reply:
Think of at as a stack. Then it is the foundation, the most stable part!